tools/mo/openvino/tools/mo/moc_frontend/check_config.py

```diff
@@ -10,17 +10,35 @@
 
 
 def any_extensions_used(argv: argparse.Namespace):
-    return hasattr(argv, 'extensions') and argv.extensions is not None and len(argv.extensions) > 0 \
-        and argv.extensions != import_extensions.default_path() # extensions arg has default value
+    if not hasattr(argv, 'extensions') or argv.extensions is None:
+        return False
+
+    if isinstance(argv.extensions, list) and len(argv.extensions) > 0:
+        has_non_default_path = False
+        has_non_str_objects = False
+        for ext in argv.extensions:
+            if not isinstance(ext, str):
+                has_non_str_objects = True
+                continue
+            if len(ext) == 0 or ext == import_extensions.default_path():
+                continue
+            has_non_default_path = True
+
+        return has_non_default_path or has_non_str_objects
+
+    raise Exception("Expected list of extensions, got {}.".format(type(argv.extensions)))
 
 
 def legacy_extensions_used(argv: argparse.Namespace):
     if any_extensions_used(argv):
-        extensions = argv.extensions.split(',')
+        extensions = argv.extensions
         legacy_ext_counter = 0
         for extension in extensions:
-            path = Path(extension)
-            if not path.is_file():
+            if not isinstance(extension, str):
+                continue
+            if extension == import_extensions.default_path():
+                continue
+            if not Path(extension).is_file():
                 legacy_ext_counter += 1
         if legacy_ext_counter == len(extensions):
```

Comment:
I think we need to extend this message `raise Error('Using new and legacy extensions in the same time is forbidden')` for a case of extension not in a string format.